app/[[...path]]/page.tsx

```diff
@@ -0,0 +1,122 @@
+import {useMemo} from 'react';
+import {getMDXComponent} from 'mdx-bundler/client';
+import {Metadata} from 'next';
+import {notFound} from 'next/navigation';
+
+import {apiCategories} from 'sentry-docs/build/resolveOpenAPI';
+import {ApiCategoryPage} from 'sentry-docs/components/apiCategoryPage';
+import {ApiPage} from 'sentry-docs/components/apiPage';
+import {DocPage} from 'sentry-docs/components/docPage';
+import {Home} from 'sentry-docs/components/home';
+import {Include} from 'sentry-docs/components/include';
+import {PlatformContent} from 'sentry-docs/components/platformContent';
+import {getDocsRootNode, nodeForPath} from 'sentry-docs/docTree';
+import {getDocsFrontMatter, getFileBySlug} from 'sentry-docs/mdx';
+import {mdxComponents} from 'sentry-docs/mdxComponents';
+import {setServerContext} from 'sentry-docs/serverContext';
+
+export async function generateStaticParams() {
+  const docs = await getDocsFrontMatter();
+  const paths = docs.map(doc => {
+    let path = doc.slug.split('/');
+    if (path[path.length - 1] === 'index') {
+      path = path.slice(0, path.length - 1);
+    }
+    return {path};
+  });
+  paths.push({path: undefined}); // the home page
+  return paths;
+}
+
+// Only render paths returned by generateStaticParams
+export const dynamicParams = false;
+export const dynamic = 'force-static';
+
+const mdxComponentsWithWrapper = mdxComponents(
+  {Include, PlatformContent},
+  ({children, frontMatter}) => <DocPage frontMatter={frontMatter}>{children}</DocPage>
+);
+
+function MDXLayoutRenderer({mdxSource, ...rest}) {
+  const MDXLayout = useMemo(() => getMDXComponent(mdxSource), [mdxSource]);
```

Comment:
I think it would be cool to properly type this function so that we can also be sure that useMemo doesn't use any dependency that has changing identity.